### ISSUE TITLE
how to get transform.txt 

### ISSUE THREAD

**Issue by Guan2014** [zlthinker/KFNet#3]
Thanks for sharing the code. I downloaded the prepared input label map files of 7scenes from the Google drive links you offered. And I tried to decorrelate the scene coordinate using ZCA algorithm. However, the Euclidean transformation matrix I got was different from yours provided in "transform.txt".  Could you tell me how you get the "transform.txt" file?

**Comment by zlthinker** [comment]
Hi,

Sorry for the late reply! To compute the "transform.txt", I first collect
the points of a full scene sequence every 10 frames. Then I offset the
points so that their coordinates will have zero mean. Finally, I apply
principle component analysis on the points and rotate them based on the
eigenvectors so that their principle axes are in alignment with the x, y, z
axes. An illustration is shown in this link: https://miro.medium.com/max/1108/1*tciJWq5IiBB9nO04AZpXGg.png, where zero-mean green points are
rotated to give blue points. The Euclidean transformation is obtained by
concatenating the translation and rotation.

Thanks!



BetterG <notifications@github.com> 于2020年4月26日周日 下午8:00写道：

**Comment by Guan2014** [comment]
@zlthinker Thanks for your reply. I compared the depth images in 7 Scenes dataset and the corresponding label map from the Google drive links you provided. I found that some depth values are not zero while their 3D coordinates are invalid(-1, -1, -1). Could you please tell me how you get the label map?

**Comment by zlthinker** [comment]
> @zlthinker Thanks for your reply. I compared the depth images in 7 Scenes dataset and the corresponding label map from the Google drive links you provided. I found that some depth values are not zero while their 3D coordinates are invalid(-1, -1, -1). Could you please tell me how you get the label map?

The depth maps provided by 7scenes dataset is acquired from the view point of depth sensors, while the depth maps that we provided are obtained by projecting the depth points into the image view point. It will cause some inconsistency of the valid/invalid regions because the depth sensor and the image sensor are associated by a fixed transformation. In order to make everything easy to use, I did the conversion so that both the depth measurements and RGB measurements lie in the same image coordinate system.
